sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -1004,11 +1051,46 @@ async def settle_messages_async(self, delivery_id: Union[int, Tuple[int, int]],
         except TypeError:
             first = delivery_id
             last = None
+
+        # Create a Message Delivery object for the Disposition
+        message_delivery = _MessageDelivery(
+            None,
+            MessageDeliveryState.WaitingToBeSent,
+            expire_time
+        )
+        on_disposition_received = partial(self._on_disposition_received_async, message_delivery)
+
         await self._link.send_disposition(
             first_delivery_id=first,
             last_delivery_id=last,
-            settled=True,
+            settled=False,
             delivery_state=state,
             batchable=batchable,
-            wait=True
+            wait=True,
+            on_disposition = on_disposition_received,
+            timeout=timeout
         )
+
+        # Wait for the message to be settled without adding more link credit
+        running = True
+        while running and message_delivery.state not in MESSAGE_DELIVERY_DONE_STATES:
+            running = await self.do_work_async(flow=False)
+
+        if message_delivery.state not in MESSAGE_DELIVERY_DONE_STATES:
+            raise MessageException(
+                condition=ErrorCondition.ClientError,
+                description="Settlement failed - connection not running."
+            )
+
+        if message_delivery.state in (
+            MessageDeliveryState.Error,
+            MessageDeliveryState.Cancelled,
+            MessageDeliveryState.Timeout
+        ):
+            try:
+                raise message_delivery.error  # pylint: disable=raising-bad-type
+            except TypeError:
+                # This is a default handler
+                raise MessageException(
```

Comment:
Do we want to raise a `MessageException` -- do we need to make a Disposition specific exception?

Reply:
ServiceBus Exception?